From 78a8c7d3d5621fe1e716604f8a2177d565f6af95 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Fri, 28 Aug 2020 17:53:24 +0200
Subject: [PATCH] Only create file URI when required

The URI is only required when a thumbnail is created or looked up.
---
 thumbnail/thumbnail.cpp | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/thumbnail/thumbnail.cpp b/thumbnail/thumbnail.cpp
index 5563e496..551f9da0 100644
--- a/thumbnail/thumbnail.cpp
+++ b/thumbnail/thumbnail.cpp
@@ -681,11 +681,9 @@ const QImage ThumbnailProtocol::getIcon()
 bool ThumbnailProtocol::createSubThumbnail(QImage& thumbnail, const QString& filePath,
                                            int segmentWidth, int segmentHeight)
 {
-    const QUrl fileUrl = QUrl::fromLocalFile(filePath);
-
-    auto getSubCreator = [&fileUrl, this]() -> ThumbCreator* {
+    auto getSubCreator = [&filePath, this]() -> ThumbCreator* {
         const QMimeDatabase db;
-        const QString subPlugin = pluginForMimeType(db.mimeTypeForUrl(fileUrl).name());
+        const QString subPlugin = pluginForMimeType(db.mimeTypeForFile(filePath).name());
         if (subPlugin.isEmpty() || !m_enabledPlugins.contains(subPlugin)) {
             return nullptr;
         }
@@ -697,8 +695,9 @@ bool ThumbnailProtocol::createSubThumbnail(QImage& thumbnail, const QString& fil
         // 128 x 128 or 256 x 256 pixels
         int cacheSize = 0;
         QCryptographicHash md5(QCryptographicHash::Md5);
-        md5.addData(fileUrl.toEncoded());
-        const QString thumbName = QFile::encodeName(md5.result().toHex()).append(".png");
+        const QByteArray fileUrl = QUrl::fromLocalFile(filePath).toEncoded();
+        md5.addData(fileUrl);
+        const QString thumbName = QString::fromLatin1(md5.result().toHex()).append(".png");
 
         if (m_thumbBasePath.isEmpty()) {
             m_thumbBasePath = QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation) + QLatin1String("/thumbnails/");
-- 
GitLab

